SPECS/mstflint/mstflint.spec

```diff
@@ -29,35 +28,114 @@ Requires:       python3
 This package contains firmware update tool, vpd dump and register dump tools
 for network adapters based on Mellanox Technologies chips.
 
-%prep
-%autosetup -p1
+%package devel
+Summary:        Scripts and header files for mstflint
+Requires:       %{name} = %{version}-1
+Provides:       mstflint-devel = %{version}-1
+
+%description devel
+Header files and scripts for mstflint
 
-find . -type f -iname '*.[ch]' -exec chmod a-x '{}' ';'
-find . -type f -iname '*.cpp' -exec chmod a-x '{}' ';'
+%prep
+%setup -q
 
 %build
-./autogen.sh
-%configure --enable-fw-mgr --enable-adb-generic-tools
+
+MSTFLINT_VERSION_STR="%{name} %{version}-%{release}"
+%configure --enable-fw-mgr --enable-adb-generic-tools MSTFLINT_VERSION_STR="${MSTFLINT_VERSION_STR}"
 %make_build
 
 %install
-%make_install
-# Remove the devel files that we don't ship
-rm -fr %{buildroot}%{_includedir}
-find %{buildroot} -type f -name "*.la" -delete -print
-find %{buildroot} -type f -name '*.a' -delete
+%make_install DESTDIR=%{buildroot}
+# remove unpackaged files from the buildroot
+rm -f %{buildroot}/%{_libdir}/*.la
+# create softlinks to old mtcr header and lib locations
+# link mtcr_ul to old location
+mkdir -p %{buildroot}/%{_includedir}/mtcr_ul
+ln -s %{_includedir}/mstflint/mtcr.h %{buildroot}/%{_includedir}/mtcr_ul/mtcr.h
+ln -s %{_includedir}/mstflint/mtcr_com_defs.h %{buildroot}/%{_includedir}/mtcr_ul/mtcr_com_defs.h
+# link mtcr_ul to old lib path
+ln -s %{_libdir}/mstflint/libmtcr_ul.a %{buildroot}/%{_libdir}/libmtcr_ul.a 
+
+%clean
+rm -fr %{buildroot}
 
 %files
-%license COPYING
-%doc README
-%{_bindir}/*
-%{_sysconfdir}/mstflint
-%{_libdir}/mstflint
-
+%defattr(-,root,root)
+%{_bindir}/mstmread
+%{_bindir}/mstmwrite
+%{_bindir}/mstflint
+%{_bindir}/mstregdump
+%{_bindir}/mstmtserver
+%{_bindir}/mstvpd
+%{_bindir}/mstmcra
+%{_bindir}/mstconfig
+%{_bindir}/mstfwreset
+%{_bindir}/mstcongestion
+%{_bindir}/mstprivhost
+%{_bindir}/mstfwtrace
+%{_bindir}/mstresourcedump
+%{_bindir}/mstresourceparse
+%{_bindir}/mstfwmanager
+%{_bindir}/mstarchive
+%{CONF_DIR}/ca-bundle.crt
+%{_bindir}/mstreg
+%{_bindir}/mstlink
+
+%files devel
+%{_includedir}/mstflint/cmdif/icmd_cif_common.h
+%{_includedir}/mstflint/cmdif/icmd_cif_open.h
+%{_includedir}/mstflint/common/compatibility.h
+%{_includedir}/mstflint/mtcr.h
+%{_includedir}/mstflint/mtcr_com_defs.h
+%{_includedir}/mstflint/mtcr_mf.h
+%{_includedir}/mstflint/tools_layouts/adb_to_c_utils.h
+%{_includedir}/mstflint/tools_layouts/icmd_layouts.h
+%{_includedir}/mtcr_ul/mtcr.h
+%{_includedir}/mtcr_ul/mtcr_com_defs.h
+%{_libdir}/mstflint/libmtcr_ul.a
+%{_libdir}/libmtcr_ul.a
+
+%{mstflint_python_tools}/tools_version.py
```

Comment:
Are these files *all* development content? The `.h` and `.a` ones seem to be, but not sure about the `.py` and `.so` files below.

Reply:
+1 , do we need the .py files for mstflint?